__init__.py

```diff
@@ -122,64 +123,104 @@ def draw(self, context):
         if text.source_text in bpy.data.texts:
             layout.prop(text,'separator')
             layout.prop(text,'text_index')
         layout.prop(text,'use_randomize')
         if text.use_randomize:
             layout.prop(text,'randomize_width')
 
+
+classes = [
+    TEXT_PT_Typewriter
+]
+
+
 def register():
     '''
     addon registration function
     '''
     # create properties
     bpy.types.TextCurve.character_start = bpy.props.IntProperty(
-      name="character_start",update=update_func, min=0, options={'ANIMATABLE'})
+        name="character_start",
+        update=update_func,
+        min=0, 
+        options={'ANIMATABLE'}
+    )
+    
+    bpy.types.TextCurve.locked = bpy.props.BoolProperty(
+        name="locked", default=True)
+    
     bpy.types.TextCurve.preserve_newline = bpy.props.BoolProperty(
-      name="preserve_newline", default=True)
+        name="preserve_newline", default=True)
+    
     bpy.types.TextCurve.preserve_space = bpy.props.BoolProperty(
       name="preserve_space", default=True)
+    
     bpy.types.TextCurve.character_count = bpy.props.IntProperty(
-      name="character_count",update=update_func, min=0, options={'ANIMATABLE'})
+        name="character_count",
+        update=update_func, 
+        min=0,
+        default=0,
+        options={'ANIMATABLE'},
+    )
+    
     bpy.types.TextCurve.backup_text = bpy.props.StringProperty(
-      name="backup_text")
+        name="backup_text")
+    
     bpy.types.TextCurve.use_animated_text = bpy.props.BoolProperty(
-      name="use_animated_text", default=False)
+        name="use_animated_text", default=False)
+    
     bpy.types.TextCurve.source_text = bpy.props.StringProperty(
-      name="source_text")
-
+        name="source_text")
 
     bpy.types.TextCurve.text_index = bpy.props.IntProperty(
-      name="index",update=update_func, min=0, options={'ANIMATABLE'})
+        name="index",
+        update=update_func, 
+        min=0, 
+        options={'ANIMATABLE'}
+    )
+      
     bpy.types.TextCurve.separator = bpy.props.StringProperty(
-      name="separator", default='#' )
+        name="separator", default='#')
+      
     bpy.types.TextCurve.use_randomize = bpy.props.BoolProperty(
-      name="randomize", default=False)
+        name="randomize", default=False)
+      
     bpy.types.TextCurve.randomize_width = bpy.props.IntProperty(
-      name="randomize width",update=update_func, default=10, min=0, options={'ANIMATABLE'})
+        name="randomize width",
+        update=update_func, 
+        default=10, 
+        min=0, 
+        options={'ANIMATABLE'}
+    )
+    
     # register the module:
-    bpy.utils.register_module(__name__)
+    from bpy.utils import register_class
+    for cls in classes:
+        register_class(cls)
+    
     # add the frame change handler
-    bpy.app.handlers.scene_update_pre.append(typewriter_text_update_frame)
+    bpy.app.handlers.frame_change_pre.append(typewriter_text_update_frame)
```

Comment:
Are the change from `scene_update_pre` to `frame_change_pre` and the change from `register_module` to `register_class` the only actual functional changes? It seems like virtually everything is style changes and README changes.

Reply:
Yes that about sums it up.